src/psyclone/tests/psyir/nodes/test_range.py

```diff
@@ -0,0 +1,211 @@
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: A. R. Porter, STFC Daresbury Lab
+
+''' pytest tests for the Range class. '''
+
+from __future__ import absolute_import
+import pytest
+from psyclone.psyir.symbols import DataType
+from psyclone.psyir.nodes import Range, Literal, Reference, Node
+from psyclone.psyGen import InternalError
+
+
+@pytest.mark.parametrize("prop", ["start", "stop", "step"])
+def test_range_getter_errors(prop):
+    ''' Test that attempting to get any of the start/stop/step properties for
+    an invalid Range object raises the expected errors. '''
+    # pylint:disable=eval-used
+    erange = Range()
+    # Manually break the list of children
+    erange._children = []
+    with pytest.raises(InternalError) as err:
+        _ = eval("erange." + prop)
+    assert ("Malformed Range: should have three children but found 0"
+            in str(err.value))
+
+    # Correct number of children but wrong type (as initialised to None)
+    erange = Range()
+    with pytest.raises(InternalError) as err:
+        _ = eval("erange." + prop)
+    assert ("Malformed Range: all children must be sub-classes of "
+            "Node" in str(err.value))
+
+
+def test_range_init(parser):
+    ''' Check that the Range constructor behaves as intended. '''
+    from fparser.common.readfortran import FortranStringReader
+    # When no arguments are provided
+    erange = Range()
+    assert erange.children == [None, None, None]
+    assert erange.parent is None
+    assert erange.annotations == []
+    assert erange.ast is None
+    # When arguments are provided
+    parent = Node()
+    reader = FortranStringReader("program hello\nend program hello\n")
+    prog = parser(reader)
+    erange2 = Range(parse_node=prog, parent=parent)
+    assert erange2.parent is parent
+    assert erange2.ast is prog
+    with pytest.raises(InternalError) as err:
+        _ = Range(annotations=["was-where"])
+    # TODO correct US spelling of unrecognized!
```

Comment:
Please either do the todo or reference an issue.

Reply:
Done it.